docs/strategies/retry.md

```diff
@@ -104,6 +104,60 @@ new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
 | `OnRetry`          | `null`                                                                     | Action executed when retry occurs.                                                       |
 | `MaxDelay`         | `null`                                                                     | Caps the calculated retry delay to a specified maximum duration.                         |
 
+## Diagrams
+
+Let's suppose we have a retry strategy with `MaxRetryAttempts`: `2`.
+
+### Happy path sequence diagram
+
+```mermaid
+%%{init: {'theme':'dark'}}%%
+sequenceDiagram
+    actor C as Caller
+    participant P as Pipeline
+    participant R as Retry
+    participant D as DecoratedUserCallback
+
+    C->>P: Calls ExecuteAsync
+    P->>R: Calls ExecuteCore
+    Note over R,D: Initial attempt
+    R->>+D: Invokes
+    D->>-R: Fails transiently
+    R->>R: Sleeps
+    Note over R,D: 1st retry attempt
+    R->>+D: Invokes
+    D->>-R: Returns result
+    R->>P: Returns result
+    P->>C: Returns result
+```
+
+### Unhappy path sequence diagram
+
+```mermaid
+%%{init: {'theme':'dark'}}%%
+sequenceDiagram
+    actor C as Caller
+    participant P as Pipeline
+    participant R as Retry
+    participant D as DecoratedUserCallback
+
+    C->>P: Calls ExecuteAsync
+    P->>R: Calls ExecuteCore
+    Note over R,D: Initial attempt
+    R->>+D: Invokes
+    D->>-R: Fails transiently
+    R->>R: Sleeps
+    Note over R,D: 1st retry attempt
+    R->>+D: Invokes
+    D->>-R: Fails transiently
+    R->>R: Sleeps
+    Note over R,D: 2nd retry attempt
```

Comment:
```suggestion
    Note over R,D: Final retry attempt
```

or `Nth`?

Reply:
I've intentionally used _2nd_ because we have made the following assumption/statement above.

> Let's suppose we have a retry strategy with `MaxRetryAttempts`: `2`.